[test-framework] agent_test.go is failing after landing to master

It looks like this test is failing in CircleCI after the collab-test-framework branch merged into master. I'll mark the test as skipped and link it to this bug.

Slack dump of conversation.

my PR https://github.com/istio/istio/pull/7750 constantly fails circleci: test
wattli
#7750 Enable permissive mode for default deployment
This change applies to istio-demo.yaml

All services will have permissive mode (they can take both http and mtls) now. But clients still send http traffic.

Will have separate PRs to remove istio-demo-auth.yaml
Labels
cla: yes
Comments
11
istio/istioAug 8thAdded by GitHub
here is more detailed info: [ENVOY-sidecar~127.0.0.1~c.eW37DwZ39Z~istio-system.svc.local] [2018-08-14 04:53:53.421][17348][trace][connection] external/envoy/source/common/network/connection_impl.cc:389] [C0] socket event: 3
2018-08-14T04:53:53.421215Z    info    ads    ADS: "127.0.0.1:40482" sidecar~127.0.0.1~b.Sbq_dF8vT3~istio-system.svc.local-2 terminated rpc error: code = Canceled desc = context canceled
2018-08-14T04:53:53.421257Z    info    rpc error: code = Canceled desc = context canceled
--- FAIL: TestAgent (20.67s)
    agent_test.go:116: failed to configure Envoys: service a missing config for cluster outbound|42951||b.istio-system.svc.local
FAIL        anyone has any ideas?
and this file agent_test.go was introduced by https://github.com/istio/istio/pull/7739/files, which was merged today
and this PR has same failure as mine https://github.com/istio/istio/pull/7893

--- a/pkg/test/framework/components/apps/local/envoy/agent/pilot/agent.go
+++ b/pkg/test/framework/components/apps/local/envoy/agent/pilot/agent.go
@@ -660,7 +660,6 @@ func (a *pilotAgent) filterDiscoveryResponse(resp *xdsapi.DiscoveryResponse) (*x
 				}
 			}
 
-			newResponse.Resources = append(newResponse.Resources, any)
 		case clusterType:
 			// Remove any management clusters.
 			c := &xdsapi.Cluster{}
@@ -670,8 +669,6 @@ func (a *pilotAgent) filterDiscoveryResponse(resp *xdsapi.DiscoveryResponse) (*x
 			switch {
 			case strings.Contains(c.Name, "mgmtCluster"):
 				continue
-			default:
-				newResponse.Resources = append(newResponse.Resources, any)
 			}
 		}
 		newResponse.Resources = append(newResponse.Resources, any)

--- a/pkg/test/framework/components/apps/local/envoy/agent/pilot/agent_test.go
+++ b/pkg/test/framework/components/apps/local/envoy/agent/pilot/agent_test.go
@@ -46,7 +46,6 @@ const (
 )
 
 func TestAgent(t *testing.T) {
-	t.Skip("https://github.com/istio/istio/issues/7910")
 	p, configStore, pilotStopFn := newPilot(namespace, t)
 	defer pilotStopFn()
 

